backend/dataall/modules/worksheets/services/worksheet_services.py

```diff
@@ -0,0 +1,220 @@
+
+    @staticmethod
+    @has_resource_permission(GET_WORKSHEET)
+    def get_worksheet(session, username, groups, uri, data=None, check_perm=None):
+        worksheet = WorksheetService.get_worksheet_by_uri(session, uri)
+        return worksheet
+
+    @staticmethod
+    @has_resource_permission(SHARE_WORKSHEET)
+    def share_worksheet(
+        session, username, groups, uri, data=None, check_perm=None
+    ) -> WorksheetShare:
+        share = WorksheetRepository.get_worksheet_share(session, uri, data)
+
+        if not share:
+            share = WorksheetShare(
+                worksheetUri=uri,
+                principalType=data['principalType'],
+                principalId=data['principalId'],
+                canEdit=data.get('canEdit', True),
+                owner=username,
+            )
+            session.add(share)
+
+            ResourcePolicy.attach_resource_policy(
+                session=session,
+                group=data['principalId'],
+                permissions=WORKSHEET_SHARED,
+                resource_uri=uri,
+                resource_type=Worksheet.__name__,
+            )
+        return share
+
+    @staticmethod
+    @has_resource_permission(SHARE_WORKSHEET)
+    def update_share_worksheet(
+        session, username, groups, uri, data=None, check_perm=None
+    ) -> WorksheetShare:
+        share: WorksheetShare = data['share']
+        share.canEdit = data['canEdit']
+        worksheet = WorksheetService.get_worksheet_by_uri(session, uri)
+        ResourcePolicy.attach_resource_policy(
+            session=session,
+            group=share.principalId,
+            permissions=WORKSHEET_SHARED,
+            resource_uri=uri,
+            resource_type=Worksheet.__name__,
+        )
+        return share
+
+    @staticmethod
+    @has_resource_permission(SHARE_WORKSHEET)
+    def delete_share_worksheet(
+        session, username, groups, uri, data=None, check_perm=None
+    ) -> bool:
+        share: WorksheetShare = data['share']
+        ResourcePolicy.delete_resource_policy(
+            session=session,
+            group=share.principalId,
+            resource_uri=uri,
+            resource_type=Worksheet.__name__,
+        )
+        session.delete(share)
+        session.commit()
+        return True
+
+    @staticmethod
+    @has_resource_permission(DELETE_WORKSHEET)
+    def delete_worksheet(
+        session, username, groups, uri, data=None, check_perm=None
+    ) -> bool:
+        worksheet = WorksheetService.get_worksheet_by_uri(session, uri)
+        session.delete(worksheet)
+        ResourcePolicy.delete_resource_policy(
+            session=session,
+            group=worksheet.SamlAdminGroupName,
+            resource_uri=uri,
+            resource_type=Worksheet.__name__,
+        )
+        return True
+
+    @staticmethod
+    def run_sql_query(session, username, groups, environmentUri, worksheetUri, sqlQuery):
+
+        ResourcePolicy.check_user_resource_permission(
```

Comment:
Shouldn't this be executed as a decorator?

Reply:
Yes, we could use `@has_resource_permission`, but we need to rename the parameter `environmentUri` to `uri`